packages/vscode-extension/src/project/metro.ts

```diff
@@ -292,6 +293,18 @@ export class Metro implements Disposable {
     ws.close();
   }
 
+  private async evaluateJsFromCdp(
+    webSocketDebuggerUrl: string,
+    source: string,
+    timeoutMs: number = 2000
+  ): Promise<string | undefined> {
+    const wsUrl = new URL(webSocketDebuggerUrl);
+    wsUrl.hostname = "localhost";
+    wsUrl.port = String(this._port);
+
+    return evaluateJsFromCdp(wsUrl.toString(), source);
```

Comment:
this function doesn't seem particularily useful and it calls a method with exact same name which is confusing. I'd just move the websocket URL resolution to `isExpoHostRuntime` where it is now called. That method is short enough to accommodate additional logic